2017/11/29/systemd-primer-from-the-trenches.html.md

```diff
@@ -0,0 +1,514 @@
+```cat /proc/net/*```
+
+Sockets of all types can be viewed from one command. Although time to time I will ise lsof or ss depending on the socket
+type I am wanting to look at.
+
+11.
+```systemctl list-timers```
+
+```crontab -e```
+
+systemd offers a way to schedule tasks aka crontab. Im going to go out on a limb here and say they both do the same thing, except
+systemd timers may be more readable by human eyes, are logged to the journal, easier to debug and enable or disable.
+
+Although my caveat; I still use cron jobs in my daily work, because Im familiar with them and emailing is still an issue from a timer.
+
+Tip:
+* set a systemd timer to a calendar, day, month, year to trigger.
+
+
+12.
+```systemctl list-jobs```
+
+Requires further explanation in another blog post.
+
+
+13.
+```systemctl --failed```
+
+Show me failed services. systemctl status will highlight at the top if units have failed, especially after a reboot.
+
+14.
+```systemctl get-default```
+
+```systemctl set-default multi-user.target``` - equivalent to runlevels 2,3 and 4
+
+to list targets run ```systemctl list-units -t target```
+
+```Runlevel```
+```Chkconfig --list```
+```telinit runlevel```
+
+Gets the run level default for the system. Not often used, if rarely, but good know when you start having boot issues or need
+to swap out to a different run level to fix things.
+
+Tip:
+* There are many tables comparing the SysV runlevels to systemd. Get your google on and search.
+* `systemctl set-default graphical.target` will set a graphical user shell. For all those that like a good desktop.
+
+15.
+```systemctl shutdown or reboot```
+
+```shutdown -r now```
+
+Reboot/shutdown the system. Personally I use ‘shutdown -r now’ still.
+
+16.
+```systemctl cat serviceName```
+
+```cd to dir, cat init file```
+
+Shows me the system service(unit) file contents and options. We will go more into these commands as we work through next week
+building and maintaining our own UNIT files.
+
+Tip:
+* systemctl cat shows all unti file information and snippets involved with unit file.
+* If you use VIM as I do, you can enable vim-systemd to help wotj syntax highlighting.
+
+17.
+```systemctl list-dependencies serviceName```
+
+What’s really depending on my process. The --all flag will show everything --before, --after, --reverse.
+
+Tip:
+* Pretty handy little dependency checker for processes.
+
+18.
+```systemctl show serviceName```
+
+Flags: -p - shows a single property of a service. 
+
+Shows more than using ‘systemctl cat servicename’. Dont forget TAB is your friend. 
+Running the ``-p`` flag and using TAB will help you.
+
+19.
+```systemctl mask serviceName```
+
+```Update-rc.d serviceName disable```
+
+Never want someone starting a service EVER!! 'Mask' is your friend and a little sneaky. See if your system admins 
+pick this one up. Good April fools day. ‘’unmask’’ to return it to its user
+
+20.
+```systemctl edit serviceName```
+
+Good flags --full
+
+```vim /etc/inittab```
```

Comment:
`/etc/inittab` isn't the same as a `systemctl edit ...` is it? It seems a better example would be editing the relevant file in `/etc/sysconfig/$whatever`. Maybe it just depends on what kind of thing it is, but I think this should have some more clearly connected example than editing `/etc/inittab`.

Reply:
changed